src/schemas/21-526EZ/schema.js

```diff
@@ -15,139 +14,118 @@ const directDepositDef = {
+ * @returns {object} the servicePeriods schema object
+ */
+const servicePeriodsDef = ((definitions) => {
+  const serviceHistory = _.cloneDeep(definitions.serviceHistory);
+  serviceHistory.minItems = 1;
+  serviceHistory.maxItems = 100;
+  serviceHistory.items.required = ['serviceBranch', 'dateRange'];
+  delete serviceHistory.items.properties.dischargeType
 
-const addressDef = {
-  type: 'object',
-  oneOf: countryStateProperties, // holds country and state
-  properties: {
-    // These validations (including regular expressions) come from the swagger docs
-    // Should these patterns start with ^ to make sure they apply to the whole string?
-    addressLine1: {
-      type: 'string',
-      maxLength: 30, // Is that going to be long enough?
-      // I think the + here means it's required
-      pattern: "([a-zA-Z0-9\-'.,,&#]([a-zA-Z0-9\-'.,,&# ])?)+$"
-    },
-    addressLine2: {
-      type: 'string',
-      maxLength: 30, // Is that going to be long enough?
-      pattern: "([a-zA-Z0-9\-'.,,&#][a-zA-Z0-9\-'.,,&# ]?)*$"
-    },
-    addressLine3: {
-      type: 'string',
-      maxLength: 30, // Is that going to be long enough?
-      pattern: "([a-zA-Z0-9\-'.,,&#][a-zA-Z0-9\-'.,,&# ]?)*$"
-    },
-    city: {
-      type: 'string',
-      maxLength: 30, // Is that going to be long enough?
-      // I think the + here means it's required
-      pattern: "([a-zA-Z0-9\-'.#]([a-zA-Z0-9\-'.# ])?)+$"
-    },
-    zipFirstFive: {
-      type: 'string',
-      // This validation isn't in the swagger docs
-      pattern: '^\d{5}$'
-    },
-    zipLastFour: {
-      type: 'string',
-      // This validation isn't in the swagger docs
-      pattern: '^\d{4}$'
-    },
-    militaryPostOfficeTypeCode: {
-      type: 'string',
-      // Should these be capitalized, or should we make sure they're lower case when we send them to the api?
-      enum: ['apo', 'dpo', 'fpo']
-    },
-    militaryStateCode: {
-      type: 'string',
-      enum: ['AA', 'AE', 'AP']
-    },
-    type: {
-      type: 'string',
-      enum: ['DOMESTIC', 'MILITARY', 'INTERNATIONAL']
-    }
-  },
-  required: ['addressLine1', 'country']
-};
+  return serviceHistory;
+})(definitions);
 
-// Apparently we don't need a suffix here.
-const fullNameDef = _.omit('properties.suffix', definitions.fullName);
+// Extracted to enable easy adding of properties for forwardingAddress
+const addressDef = definitions.pciuAddress;
 
-// Not sure this particular kind of datetime definition will be used anywhere else
-// Pattern matches datetimes like 2018-03-22T17:25:19.191Z where the fractional seconds are optional
-// NOTE: This doesn't catch invalid days like February 30th
-const datetime = {
-  pattern: '(\d{4}|XXXX)-(0[1-9]|1[0-2]|XX)-(0[1-9]|[1-2][0-9]|3[0-1]|XX)T([01]\d|2[0-3]):[0-5]\d:[0-5]\d(\.\d+)?Z',
-  type: 'string'
-}
+// Strip suffix from fullName common def
+const fullNameDef = _.merge(definitions.fullName, {
+  properties: _.omit('suffix', definitions.fullName.properties)
```

Comment:
Oh, right. Because `_.omit()` doesn't accept a path. :man_facepalming:
Good catch.

Reply:
Is this being based off alternateName?

Reply:
Right, yeah. That's the only name field in the form (which still seems odd to me).

Reply:
Yeah, that's the only usage. Though checking swagger, it looks as though we don't have the right regex patterns set up to validate against. Will add.